src/liboslexec/llvm_gen.cpp

```diff
@@ -2168,11 +2177,12 @@ LLVMGEN (llvm_gen_texture)
     args.push_back (rop.ll.void_ptr (rop.llvm_get_pointer (Result, 0)));
     args.push_back (rop.ll.void_ptr (rop.llvm_get_pointer (Result, 1)));
     args.push_back (rop.ll.void_ptr (rop.llvm_get_pointer (Result, 2)));
-    if (alpha) {
-        args.push_back (rop.ll.void_ptr (alpha));
+    if (alpha || texture_handle) {
+        args.push_back (rop.ll.void_ptr (alpha    ? alpha    : rop.ll.void_ptr_null()));
         args.push_back (rop.ll.void_ptr (dalphadx ? dalphadx : rop.ll.void_ptr_null()));
         args.push_back (rop.ll.void_ptr (dalphady ? dalphady : rop.ll.void_ptr_null()));
-        rop.ll.call_function ("osl_texture_alpha", &args[0], (int)args.size());
+        rop.ll.call_function (texture_handle ? "osl_texture_handle" : "osl_texture_alpha",
```

Comment:
Yes, osl_texture and osl_texture_alpha should be merged. I decided not to replicate that unnecessary split for the handle routines, and I want to come back and merge the redundancies. Should I do it as part of the same review?